tiled/src/main/kotlin/ktx/tiled/mapLayer.kt

```diff
@@ -0,0 +1,38 @@
+package ktx.tiled
+
+import com.badlogic.gdx.maps.MapLayer
+import com.badlogic.gdx.maps.MapProperties
+
+/**
+ * Extension method to directly access the [MapProperties] of a [MapLayer]. If the property
+ * is not defined then this method throws a [MissingPropertyException].
+ * @param key property name
+ * @return value of the property
+ * @throws MissingPropertyException If the property is not defined
+ */
+inline fun <reified T> MapLayer.property(key: String): T = this.properties[key, T::class.java]
+        ?: throw MissingPropertyException("Property $key does not exist for layer ${this.name}")
+
+/**
+ * Extension method to directly access the [MapProperties] of a [MapLayer]. The type is automatically
+ * derived from the type of the given default value. If the property is not defined the defaultValue will be returned.
+ * @param key property name
+ * @param defaultValue default value in case the property is missing
+ * @return value of the property or defaultValue if property is missing
+ */
+inline fun <reified T> MapLayer.property(key: String, defaultValue: T): T = this.properties[key, defaultValue, T::class.java]
```

Comment:
Can this call return a null? E.g. if the key is linked to an existing property with `null` value. It would make more sense to throw MissingPropertyException than receive a NPE. The same applies to other such calls.

Reply:
In my opinion it cannot do that and it is also part of the Tests that I wrote. When checking the original method, from what I can tell, it will always return the defaultValue if either the key is missing or the linked value to the key is null.

```Java
public <T> T get (String key, T defaultValue, Class<T> clazz) {
	Object object = get(key);
	return object == null ? defaultValue : (T)object;
}
```

Reply:
OK, thanks for verifitcation.